client/balrogclient/api.py

```diff
@@ -118,12 +119,17 @@ def do_request(self, url, data, method):
             logging.debug('Data sent: %s', sanitised_data)
         else:
             logging.debug('Data sent: %s', data)
+        data = json.dumps(data)
+        if method == 'PUT' or method == 'POST':
+            data = data.encode('utf-8')
+            data = gzip.compress(data)
```

Comment:
Sorry, but an unstated requirement is that this should run on python 2.7 and this is only in 3.2+.